### PR TITLE
Add HPU Activity Type

### PR DESCRIPTION
**Pull request by astachowiczhabana**
Summary: This change is part of Habana Device enablement in PyTorch.
    Allows user to select HPU as profiling activity.

Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by robieta** [comment]
LGTM. Once the corresponding PRs in https://github.com/pytorch/kineto are approved and we update the pinned version I'll get this going.

**Comment by astachowiczhabana** [comment]
Thank you @robieta PR in kineto is: https://github.com/pytorch/kineto/pull/611

**Comment by astachowiczhabana** [comment]
Hi @robieta. The corresponding PR for Kineto is now merged (https://github.com/pytorch/kineto/pull/611) can we also merge this one?

File: torch/csrc/profiler/kineto_shim.cpp [hunk 1]
```diff
@@ -31,6 +31,10 @@ const std::set<libkineto::ActivityType> cudaTypes = {
     // CUDA_RUNTIME appears in both cpuTypes and cudaTypes.
     libkineto::ActivityType::CUDA_RUNTIME,
 };
+
+const std::set<libkineto::ActivityType> hpuTypes = {
+    libkineto::ActivityType::HPU_OP,
+  };
```
Comment (robieta) [inline torch/csrc/profiler/kineto_shim.cpp:37]: FYI Clang format also wants you to fix this indentation.

File: torch/csrc/profiler/kineto_shim.cpp [hunk 2]
```diff
@@ -296,6 +303,8 @@ c10::DeviceType deviceTypeFromActivity(libkineto::ActivityType activity_type) {
     case libkineto::ActivityType::GLOW_RUNTIME:
     case libkineto::ActivityType::PYTHON_FUNCTION:
       return c10::DeviceType::CPU;
+    case libkineto::ActivityType::HPU_OP:
+        return c10::DeviceType::HPU;
```
Comment (robieta) [inline torch/csrc/profiler/kineto_shim.cpp:307]: And this one

File: torch/csrc/profiler/api.h [hunk 3]
```diff
@@ -15,6 +15,7 @@ namespace impl {
 enum class C10_API_ENUM ActivityType {
   CPU = 0,
   CUDA, // CUDA kernels, runtime
+  HPU,  // HPU kernels
```
Comment (robieta) [inline torch/csrc/profiler/api.h:18]: I think you have to have exactly one space between the comma and comment. I'm not 100% sure; clang format is complaining but is tight lipped about why.

**Comment by robieta** [comment]
All the builds are going to fail until https://github.com/pytorch/pytorch/pull/80974 is merged. We will need to handle https://github.com/pytorch/kineto/pull/628 before that can happen.

**Review by robieta — APPROVED** [review]
Approving. Will merge once CI is passing.

**Comment by astachowiczhabana** [comment]
Hi @robieta. Merge conflicts and lint issues are fixed. CI won't pass until Kineto submodule is updated to this commit: https://github.com/pytorch/kineto/commit/115f530612ce668f88fc966477e5afdc690679de. Can you please help with this?

**Pull request closed without merging**